src/PIL/DdsImagePlugin.py

```diff
@@ -148,15 +147,38 @@ def _open(self):
             self.tile = [("raw", (0, 0) + self.size, 0, (self.mode, 0, 1))]
         elif pfflags & DDPF_RGB:
             # Texture contains uncompressed RGB data
-            masks = {mask: ["R", "G", "B", "A"][i] for i, mask in enumerate(masks)}
-            rawmode = ""
-            if pfflags & DDPF_ALPHAPIXELS:
-                rawmode += masks[0xFF000000]
-            else:
+            if not pfflags & DDPF_ALPHAPIXELS:
                 self._mode = "RGB"
-            rawmode += masks[0xFF0000] + masks[0xFF00] + masks[0xFF]
 
-            self.tile = [("raw", (0, 0) + self.size, 0, (rawmode[::-1], 0, 1))]
+            # Each channel is derived using a mask of the pixel values
+            masks = {
+                mask: "RGB"[i]
+                for i, mask in enumerate(struct.unpack("<3I", header.read(12)))
+            }
+            if self.mode == "RGB" and all(
+                mask in (0b0000000000011111, 0b0000001111100000, 0b0111110000000000)
+                for mask in masks
+            ):
+                # Each mask uses 5 bits
+                rawmode = (
+                    masks[0b0000000000011111]
+                    + masks[0b0000001111100000]
+                    + masks[0b0111110000000000]
+                    + ";15"
+                )
+            else:
+                # Try 8 bits for each mask
```

Comment:
What's the trying here? Rather, what happens if trying fails?

Reply:
Given that the issue related to this issue has found 5 bit masks, it seems possible that there could be other mask values used in files, beyond 5 bit and 8 bits.

If it fails, an error is raised, and ultimately an `UnidentifiedImageError`.